### PR TITLE
Add Getting Started and Integration guides

### PR DESCRIPTION
**Pull request by b-onc**
Co-authored-by: Matheus Cardoso <matheus@cardo.so>

### CODE REVIEW

File: docusaurus/docs/iOS/guides/getting-started.md [hunk 1]
```diff
@@ -1,3 +1,308 @@
...
+chatClient
+  .channelController(for: <ChannelId>)
+  .createNewMessage(text: "My first message")
+```
+
+#### Controllers can be used for continuous observations of the given object
+
+Controllers can also act as entity observers and monitor changes of the represented entity. There's no limitation in terms of how many controllers can observe the same entity.
+
+You can choose the preferred way you want to be notified about the changes:
```
Comment (nikomancy) [inline docusaurus/docs/iOS/guides/getting-started.md:48]: We should try not to present more options than necessary in getting started. Are there specific use cases for these different approaches? 

If yes, could you add a line or two summarizing when / why to use one of the example approaches over another.

File: docusaurus/docs/iOS/guides/getting-started.md [hunk 2]
```diff
@@ -1,3 +1,308 @@
...
+
+User Tokens are JWT tokens containing a User ID and used to authenticate a user. You can [generate these tokens using a server client](https://getstream.io/chat/docs/ios-swift/tokens_and_authentication/?language=swift#generating-tokens) and then use them to authenticate a user in a chat client.
+
+:::note
+Authenticating a user with a valid user token creates the said user in Stream backend if it's not created yet.
+
+To create a new user, simply create a user token for the desired user id and connect with the said user id.
+:::
+
+### Regular User
```
Comment (nikomancy) [inline docusaurus/docs/iOS/guides/getting-started.md:155]: Let's have the getting started also introduce enabling developer tokens and assume their use as the default. We can provide other options further into the guide or have links to the CMS section on it the way you currently do.

Comment (b-onc) [reply docusaurus/docs/iOS/guides/getting-started.md:155]: I need some clarification on this... don't we want to show other scenarios in the guide?

File: docusaurus/docs/iOS/guides/getting-started.md (hunk 1 above)
Comment (b-onc) [reply docusaurus/docs/iOS/guides/getting-started.md:48]: It's actually pretty clear for iOS devs but I'm adding a couple of sentences.

**Review by nikomancy — APPROVED** [review]
This should be good enough to start.

File: docusaurus/docs/iOS/guides/working-with-user.md [hunk 3]
```diff
@@ -1,3 +1,185 @@
...
+
+## CurrentUser vs User
+
+You can get 2 types of controllers from `ChatClient`: `CurrentUserController` vs `UserController`
+
+### `CurrentUserController`
+
+`CurrentUser` is a wrapper for the currently logged-in user's `User` object. `CurrentUser` includes information about the logged-in user, such as registered devices, muted users, flagged users, and unread count. You interact with your current user via `CurrentUserController`
+
+For all functions available on this controller, please check [CurrentUserController docs](404).
```
Comment (DominikBucher12) [inline docusaurus/docs/iOS/guides/working-with-user.md:144]: Don't forget to add the reference docs link after rebase:

`../ReferenceDocs/Sources/StreamChat/Controllers/CurrentUserController/CurrentUserController` (double check if this is right).

File: docusaurus/docs/iOS/guides/working-with-user.md [hunk 4]
```diff
@@ -1,3 +1,185 @@
...
+    .map(\.messages)
+    .sink { UIApplication.shared.applicationIconBadgeNumber = $0 }
+    .store(in: &cancellables)
+```
+
+### `UserController`
+
+For any user other than your current user in the platform, you use `UserController` to interact.
+
+To see available functions on UserContorller, please check [UserController guide](404).
```
Comment (DominikBucher12) [inline docusaurus/docs/iOS/guides/working-with-user.md:185]: Another 404 just mentioning + typo: `UserContorller` -> `UserController`

**Review by DominikBucher12 — APPROVED** [review]
I love the tables (classic tint x pink tint, darkmode x lightmode etc.)  in getting started 😍 
Just small reminders + fix typo on last line 

and then :shipit:

**Pull request merged**